api/lib/read-graphs/source.ts

```diff
@@ -21,7 +21,9 @@ handle<SourceEvents, 'SourceUploaded'>('SourceUploaded', ev => {
       a dataCube:Source, dataCube:CsvSource ;
       dataCube:project <${ev.data.projectId}> ;
       schema:name "${ev.data.fileName}" ;
-      api:columns <${ev.id}/columns> .
+      api:columns <${ev.id}/columns> ;
+      dataCube:csvDelimiter """${ev.data.delimiter}""" ;
+      dataCube:csvQuote "${ev.data.quote.replace('"', '\\"')}" .
```

Comment:
This manual escaping made me wonder: is this the only character that could cause issue? Is there an existing function to escape strings for RDF literals?

Reply:
Those are good questions indeed. It does seem that the only character that needs to be escaped is indeed that character which is used to wrap the string value. Both single and double quote is valid and SPARQL also has the triple-quote construct for multiline text and text with quotations. 

Sadly, `'''''''` is not valid (that is seven single quote chars) but this is: `''' ' '''` 🤷‍♂ 

I have no better idea than simply escape the quote

Reply:
There is an escape function here https://github.com/rdfjs-base/to-ntriples/blob/master/lib/literal.js, do you already have that in your dependencies?

Reply:
Yes. can use that. But then, honestly, this should be used for all string values in SPARQL.

Best if we finally came up with something better than string concatenation.

Reply:
> Best if we finally came up with something better than string concatenation.

I have nothing to add to that ;)